cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -1096,6 +1096,81 @@ public void testSchedules() throws Exception {
     testUpdateSchedule(appV2Id);
   }
 
+  @Test
+  public void testUpdateSchedulesFlag() throws Exception {
+    // deploy an app with schedule
+    AppWithSchedule.AppConfig config = new AppWithSchedule.AppConfig(true, true, true);
+
+    Id.Artifact artifactId = Id.Artifact.from(TEST_NAMESPACE_META2.getNamespaceId().toId(),
+                                              AppWithSchedule.NAME, VERSION1);
+    addAppArtifact(artifactId, AppWithSchedule.class);
+    AppRequest<? extends Config> request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, true);
+
+    ApplicationId defaultAppId = TEST_NAMESPACE_META2.getNamespaceId().app(AppWithSchedule.NAME);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    List<ScheduleSpecification> actualSchSpecs = listSchedules(TEST_NAMESPACE_META2.getNamespaceId().getNamespace(),
+                                                               defaultAppId.getApplication(),
+                                                               defaultAppId.getVersion());
+    Assert.assertEquals(2, actualSchSpecs.size());
+
+    // with workflow, without schedule
+    config = new AppWithSchedule.AppConfig(true, false, false);
+    request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, false);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    // schedule should not be updated
```

Comment:
We set scheduleUpdate flag to be and are now deploying with deleted schedules. Shouldn't this update the schedule i.e. delete them ? I don't think workflow existence should affect this because now with app deployment user does not have a way delete existing schedule.